File: src/main/java/com/seaweed/identifier/SessionContext.java
package com.seaweed.identifier;

import com.seaweed.identifier.login.LoginVO;
import com.seaweed.identifier.user.vo.UserVO;
import jakarta.servlet.http.HttpSession;
import org.springframework.web.context.request.RequestContextHolder;
import org.springframework.web.context.request.ServletRequestAttributes;

public class SessionContext {
    public static HttpSession getSession(){
        return ((ServletRequestAttributes) RequestContextHolder.currentRequestAttributes()).getRequest().getSession();
    }
    public static LoginVO getAuth(){
        return (LoginVO)getSession().getAttribute("auth");
    }

}

File: src/main/kotlin/com/seaweed/identifier/SessionContext.kt
package com.seaweed.identifier

import com.seaweed.identifier.user.User
import jakarta.servlet.http.HttpSession
import org.springframework.web.context.request.RequestContextHolder
import org.springframework.web.context.request.ServletRequestAttributes

class SessionContext {
    companion object{
        fun getSession() : HttpSession {
            return (RequestContextHolder.currentRequestAttributes() as ServletRequestAttributes).request.getSession()
        }
        fun getAuth() : User {
            return getSession().getAttribute("auth") as User
        }

    }
}